pilot/pkg/xds/sds.go

```diff
@@ -96,7 +96,11 @@ func (s *SecretGen) parseResources(names []string, proxy *model.Proxy) []SecretR
 		pkpConfHashStr = strconv.FormatUint(xxhashv2.Sum64String(pkpConf.String()), 10)
 	}
 	for _, resource := range names {
-		sr, err := credentials.ParseResourceName(resource, proxy.VerifiedIdentity.Namespace, proxy.Metadata.ClusterID, s.configCluster)
+		secretCluster := proxy.Metadata.ClusterID
+		if !features.CentralIstiodAccess {
+			secretCluster = s.configCluster // ignore CLUSTER_ID header, only secrets from Istiod's cluster
```

Comment:
It is much clear using the flag here. But not on the authz on xds connection. I think the authz on xds conn should not couple with the secret gen

Reply:
Agreed, I came to same conclusion - latest change is splitting this in 2 toggles.

Do you think we can default the cross-cluster secret access to false ?

Reply:
Yes, to keep consistent with previous releases

Reply:
Previous releases did allow cross-cluster secret by default - including 'istiod per cluster'. It is not a documented feature and it doesn't seem to have any test ( since setting it to false is passing the test suites ), and since secret-access blast radius is the biggest risk I would keep it false, as a security fix instead of backward compat. 

There are few users of 'central istiod' and they tend to be sophisticated enough to read the release notes and make changes if they need ( it only affects DestinationRule secrets AFAIK )